src/operation/emulation_model_optimizer_results.jl

```diff
@@ -0,0 +1,71 @@
+"""
+Stores results data for one EmulationModel
+"""
+mutable struct EmulationModelOptimizerResults <: AbstractModelOptimizerResults
+    last_recorded_row::Int
+    duals::Dict{ConstraintKey, DataFrames.DataFrame}
+    parameters::Dict{ParameterKey, DataFrames.DataFrame}
+    variables::Dict{VariableKey, DataFrames.DataFrame}
+    aux_variables::Dict{AuxVarKey, DataFrames.DataFrame}
+    expressions::Dict{ExpressionKey, DataFrames.DataFrame}
+    optimizer_stats::OrderedDict{Int, OptimizerStats}
+end
+
+function EmulationModelOptimizerResults()
+    return EmulationModelOptimizerResults(
+        0,
+        Dict{ConstraintKey, DataFrames.DataFrame}(),
+        Dict{ParameterKey, DataFrames.DataFrame}(),
+        Dict{VariableKey, DataFrames.DataFrame}(),
+        Dict{AuxVarKey, DataFrames.DataFrame}(),
+        Dict{ExpressionKey, DataFrames.DataFrame}(),
+        OrderedDict{Int, OptimizerStats}(),
+    )
+end
+
+function write_result!(
+    data::EmulationModelOptimizerResults,
+    field::Symbol,
+    key::OptimizationContainerKey,
+    execution::Int,
+    array,
+    columns,
+)
+    container = getfield(data, field)
+    df = axis_array_to_dataframe(array, columns)
+    container[key][execution, :] = df[1, :]
+    return
+end
+
+function read_results(
+    data::EmulationModelOptimizerResults,
+    container_type::Symbol,
+    key::OptimizationContainerKey,
+    index = nothing,
+)
+    container = getfield(data, container_type)
+    # Return a copy because callers may mutate it.
+    return copy(container[key], copycols = true)
+end
+
+function write_optimizer_stats!(
+    results::EmulationModelOptimizerResults,
+    stats::OptimizerStats,
+    execution::Int,
+)
+    # TODO DT: This trips in one test. Should we enforce this rule?
```

Comment:
We should enable it and then confirm why it happens.

Reply:
We weren't clearing the model stores. Fixed.